scc/light_client/retry.go

```diff
@@ -0,0 +1,115 @@
+package light_client
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/0xsoniclabs/sonic/scc"
+	"github.com/0xsoniclabs/sonic/scc/cert"
+	"github.com/Fantom-foundation/lachesis-base/inter/idx"
+)
+
+// retryP is a provider that retries requests a certain number of times on failed
+// requests for certificates with a determined wait between retries. It is used
+// to wrap other providers to add a retry mechanism.
+//
+// Fields:
+// - provider: The underlying provider to retry requests on.
+// - attempts: The maximum number of attempt to ask for certificates.
+// - delay: The time to wait between retries.
+type retryP struct {
```

Comment:
nit: no appropriations; we have enough storage capacity and horizontal screen space to use full names like `retryProvider`.

Reply:
https://github.com/0xsoniclabs/sonic/pull/128/commits/91dfbe1c5d1a763e2d73c147f8e929977c9b9f00